### PR TITLE
CI: Fix e2e flakes

### PR DESCRIPTION
**Pull request by eloycoto**
With the new changes on RPM/systemd/sysusers the e2e was failing hard.
This commit adds the following changes:

- Avoid the use of machine-ctl and use sudo -u flotta
- Forcing endpoint container to have CgroupNsMode to private.
- Separate kind phases to make it easier to debug failures.
- Gatter better logs

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ var _ = Describe("e2e", func() {
 			Expect(stdout).To(Equal("active"))
 
 			// Check the podman.socket is running:
-			stdout, err = device.Exec("systemctl --machine flotta@.host is-active --user podman.socket")
```
Comment (jordigilh) [inline test/e2e/e2e_test.go:73]: I'm curious, why the change?

**Review by jordigilh — APPROVED** [review]

File: test/e2e/e2e_test.go (hunk 1 above)
Comment (eloycoto) [reply test/e2e/e2e_test.go:73]: mainly, because all systemd env vars that are needed, so checking as flotta user using sudo it's much simpler.

Comment (machacekondra) [reply test/e2e/e2e_test.go:73]: Strange that the command don't fail on missing env vars. I thought sudo pam modules don't create the needed env vars.
